### PR TITLE
Add has_image filter to document search (#691)

### PR DESCRIPTION
**Pull request by blms**
## What this PR does

- Allows the user to filter documents by the presence of IIIF images

### CODE REVIEW

File: geniza/corpus/forms.py [hunk 1]
```diff
@@ -200,6 +200,10 @@ class DocumentSearchForm(forms.Form):
         # Translators: label for document type search form filter
         label=_("Document Type"),
     )
+    has_image = BooleanFacetField(
+        # Translators: label for "has transcription" search form filter
```
Comment (rlskoeser) [inline geniza/corpus/forms.py:204]: looks like you copied the comment and forgot to update it :-) 

```suggestion
        # Translators: label for "has image" search form filter
```

**Review by rlskoeser — CHANGES_REQUESTED** [review]
Looks good — it's great that we're at a point where it's so easy to add new filters like this.

I did notice that the filter icon didn't change on the site — looks like you need to update the `filters_active` method. Should we add a comment to the part of the code where new form fields would be added as a reminder to update that method? OR, alternatively, could we consider everything except keyword search and sort to be filters and write it that way, so that we won't have to update it every time we add a new filter? (that option sounds better to me, but I could see it being a separate refactor — if you agree, you could make a new issue for that and do the simple fix now).

I'm not sure about the filter order — I don't think I would have put images first (I think scholarship records are more important). But probably best to get input from Rachel on that, and easy enough to switch order if we need to.

File: geniza/corpus/forms.py [hunk 2]
```diff
@@ -226,13 +226,15 @@ class DocumentSearchForm(forms.Form):
...
-                or self.cleaned_data["has_transcription"]
-                or self.cleaned_data["has_translation"]
-                or self.cleaned_data["has_discussion"]
+            filter_fields = list(
+                filter(lambda k: k not in ["q", "sort"], self.fields.keys())
             )
+            active_fields = dict(
+                filter(lambda i: bool(i[1]), self.cleaned_data.items())
+            ).keys()
+            return bool(set(filter_fields).intersection(set(active_fields)))
```
Comment (rlskoeser) [inline geniza/corpus/forms.py:237]: This looks more complicated than I think we need (and it's hard to read).

How about this?

```suggestion
            return bool(
                {
                    k: v
                    for k, v in self.cleaned_data.items()
                    if k not in ["q", "sort"] and bool(v)
                }
            )
```

File: geniza/corpus/forms.py [hunk 3]
```diff
@@ -226,13 +226,15 @@ class DocumentSearchForm(forms.Form):
     }
 
     def filters_active(self):
+        # returns True if any filter fields are active in cleaned_data
```
Comment (rlskoeser) [inline geniza/corpus/forms.py:229]: we should probably update the comment here (and should probably be a docstring, too)

```suggestion
        """Check if any filters are active; returns true if form fields other than sort or q are set"""
```

File: geniza/corpus/forms.py (hunk 2 above)
Comment (blms) [reply geniza/corpus/forms.py]: Looks much cleaner! Didn't realize we didn't need to actually compare it with `self.fields`, great solution.

**Review by rlskoeser — APPROVED** [review]

**Pull request merged**